internal/event-collector/ykclient/client.go

```diff
@@ -0,0 +1,187 @@
+		err = json.Unmarshal(line, &ev)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "could not unmarshal event from stream: %v\n", err)
+			break
+		}
+
+		if ev.Type == si.EventRecord_APP {
+			fmt.Printf("Application\n")
+			fmt.Printf("---------\n")
+			fmt.Printf("Type         : %s\n", si.EventRecord_Type_name[int32(ev.Type)])
+			fmt.Printf("ObjectId     : %s\n", ev.ObjectID)
+			fmt.Printf("Message      : %s\n", ev.Message)
+			fmt.Printf("Change Type  : %s\n", ev.EventChangeType)
+			fmt.Printf("Change Detail: %s\n", ev.EventChangeDetail)
+			fmt.Printf("Reference ID:  %s\n", ev.ReferenceID)
+		}
+	}
+	return nil
+}
+
+func (c *Client) loadUpCurrentClusterState() error {
+	partitions, err := c.loadCurrentPartitions()
+	_, err = c.loadCurrentQueues(partitions)
+	// ... other resources
+	return err
+}
+
+func (c *Client) loadCurrentPartitions() ([]*dao.PartitionInfo, error) {
+	partitions := []*dao.PartitionInfo{}
+	url := c.endPointURL(partitionsEndPt)
+	resp, err := http.Get(url)
+	if err != nil {
+		return nil, fmt.Errorf("could not get partitions from %s: %v", url, err)
+	}
+	reader := bufio.NewReader(resp.Body)
+	for {
+		line, err := reader.ReadBytes('\n')
+		if err != nil {
+			if err == io.EOF {
+				break
+			}
+			return nil, fmt.Errorf("could not read partitions from response: %v", err)
+		}
+		if len(line) == 0 {
+			break
+		}
+		pi := []*dao.PartitionInfo{}
+		err = json.Unmarshal(line, &pi)
+		if err != nil {
+			return nil, fmt.Errorf("could not unmarshal partition from response: %v", err)
+		}
+		partitions = append(partitions, pi...)
+	}
+	// Insert partitions into DB
+	for _, p := range partitions {
+		// Insert partition into DB
+		_, err := c.repo.Dbpool.Exec(context.Background(), "INSERT INTO partitions (id, cluster_id, name, capacity, used_capacity, utilization, total_nodes, applications, total_containers, state, last_state_transition_time) VALUES ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11)",
+			uuid.New(), p.ClusterID, p.Name, p.Capacity.Capacity, p.Capacity.UsedCapacity, p.Capacity.Utilization, p.TotalNodes, p.Applications, p.TotalContainers, p.State, p.LastStateTransitionTime)
+		if err != nil {
+			return nil, fmt.Errorf("could not insert partition into DB: %v", err)
+		}
+	}
+	return partitions, nil
+}
+
+func (c *Client) loadCurrentQueues(partitions []*dao.PartitionInfo) (map[string][]*dao.PartitionQueueDAOInfo, error) {
+	queuesByPartition := map[string][]*dao.PartitionQueueDAOInfo{}
+	for _, p := range partitions {
+		url := c.endPointURL(queuesEndPt(p.Name))
+
+		resp, err := http.Get(url)
+		if err != nil {
+			return nil, fmt.Errorf("could not get partitions from %s: %v", url, err)
+		}
+		reader := bufio.NewReader(resp.Body)
+		for {
+			line, err := reader.ReadBytes('\n')
+			if err != nil {
+				if err == io.EOF {
+					break
+				}
+				return nil, fmt.Errorf("could not read queues from response: %v", err)
+			}
+			if len(line) == 0 {
+				break
+			}
+			qi := dao.PartitionQueueDAOInfo{}
```

Comment:
this is not an array because the response returns a single `PartitionQueueDAOInfo` json object because I tested it with a single queue, not sure what type will it returns when there are multiple queues. while [here](https://github.com/richscott/yhs/pull/3/files#diff-34710ffd86fb0b0cf6ccc134602797363742d294f572840cbc5c98cbde26116cR124) it returns an json array.